src/LoggerServiceFactory.php

```diff
@@ -34,9 +44,108 @@ public function __invoke(ContainerInterface $container, $name, ?array $options =
      * Proxies to `__invoke()`.
      *
      * @return Logger
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
      */
     public function createService(ServiceLocatorInterface $serviceLocator)
     {
         return $this($serviceLocator, Logger::class);
     }
+
+    /**
+     * Process and return the configuration from the container.
+     *
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    protected function processConfig(array &$config, ContainerInterface $services)
+    {
+        if (
+            isset($config['writer_plugin_manager'])
+            && is_string($config['writer_plugin_manager'])
+            && $services->has($config['writer_plugin_manager'])
+        ) {
+            $config['writer_plugin_manager'] = $services->get($config['writer_plugin_manager']);
+        }
+
+        if (
+            (! isset($config['writer_plugin_manager'])
+                || ! $config['writer_plugin_manager'] instanceof AbstractPluginManager)
+            && $services->has('LogWriterManager')
+        ) {
+            $config['writer_plugin_manager'] = $services->get('LogWriterManager');
+        }
+
+        if (
+            isset($config['processor_plugin_manager'])
+            && is_string($config['processor_plugin_manager'])
+            && $services->has($config['processor_plugin_manager'])
+        ) {
+            $config['processor_plugin_manager'] = $services->get($config['processor_plugin_manager']);
+        }
+
+        if (
+            (! isset($config['processor_plugin_manager'])
+                || ! $config['processor_plugin_manager'] instanceof AbstractPluginManager)
+            && $services->has('LogProcessorManager')
+        ) {
+            $config['processor_plugin_manager'] = $services->get('LogProcessorManager');
+        }
+
+        if (! isset($config['writers'])) {
+            return;
+        }
+
+        foreach ($config['writers'] as $index => $writerConfig) {
+            if (
+                isset($writerConfig['name'])
```

Comment:
We're making the assumption that `$writerConfig` is an array or `ArrayAccess` implementation; should we assert before this that this is the case? It might help identify incorrect configuration earlier.

Reply:
added guard statement